File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var nums = number.ToString().Select(c => int.Parse(c.ToString())).ToArray();
        return nums.Select(n => Math.Pow(n, nums.Length)).Sum() == number;
    }
}

File: go/armstrong-numbers/armstrong_numbers.go
package armstrong

import (
	"math"
	"strconv"
)

func IsNumber(n int) bool {
	sum := 0
	s := strconv.Itoa(n)
	for _, c := range s {
		r := math.Pow(float64(c-'0'), float64(len(s)))
		sum += int(r)
	}
	return sum == n
}
